ingestion/ca/open_gov_canada.py

```diff
@@ -0,0 +1,183 @@
+        description=ld_record["notes"],
+        alternate_names=[
+            AltStr(value=ld_record["title_translated"][lang], lang=lang)
+            for lang in ld_record["title_translated"].keys()
+        ],
+        alternate_descriptions=[
+            AltStr(value=ld_record["notes_translated"][lang], lang=lang)
+            for lang in ld_record["notes_translated"].keys()
+        ],
+        upstream_id=ld_record["id"],
+        source_url=pd.url,
+        upstream_upload_time=parse_date(ld_record["date_published"]),
+        license=ld_record["license_title"],
+        # subject & topic_category fields are lists, keywords is dict with lists for each language
+        tags=ld_record["subject"]
+        + ld_record.get("topic_category", [])
+        + reduce(
+            lambda x, y: x + y,
+            [ld_record["keywords"][lang] for lang in ld_record["keywords"].keys()],
+        ),
+        publisher_name=ld_record["organization"]["title"],
+        publisher_url=SITE_URL,
+        region_name="Canada",
+        subregion=provinces.get(ld_record["organization"]["name"], ""),
+        region_country_code="ca",
+    )
+    for file in ld_record["resources"]:
+        ds.add_file(
+            url=file["url"], file_type=handle_file_format(file["format"]), name=file["name"]
+        )
+
+    return ds
+
+
+# handle file formats:
+# TO DO: complete list of file formats
+def handle_file_format(original_format):
+    format = original_format.lower()
+    if format in APPROVED_FORMATS:
+        return format
+    elif format in ["sql", "sql lite"]:
+        return "sqlite"
+    elif format in OTHER_GEO_FORMATS:
+        return "other-geo"
+    # elif format in ["docx", "edi", "html", "jpg", "kmz", "pdf", "rdf", "rss", "tiff", "zip"]:
+    #    return "other"
+    else:
+        return "other"
+
+
+# Short codes used in organization->name for provincial govts (and territories)
+ALBERTA = "ab"
+BRITISH_COLUMBIA = "bc-cb"
+MANITOBA = "mb"
+NEW_BRUNSWICK = "nb"
+NEWFOUNDLAND_LABRADOR = "nl-tnl"
+NORTHWEST_TERRITORIES = "nwt-tno"
+NOVA_SCOTIA = "ns-ne"
+ONTARIO = "on"
+PRINCE_EDWARD = "pei-ipe"
+QUEBEC = "qc"
+SASKATCHEWAN = "sk"
+YUKON = "yk"
+
+# Dictionary of Canadian provinces/territories
+provinces = {
+    ALBERTA: "Alberta",
+    BRITISH_COLUMBIA: "British Columbia",
+    MANITOBA: "Manitoba",
+    NEW_BRUNSWICK: "New Brunswick",
+    NEWFOUNDLAND_LABRADOR: "Newfoundland and Labrador",
+    NORTHWEST_TERRITORIES: "Northwest Territories",
+    NOVA_SCOTIA: "Nova Scotia",
+    ONTARIO: "Ontario",
+    PRINCE_EDWARD: "Prince Edward Island",
+    QUEBEC: "Québec",
+    SASKATCHEWAN: "Saskatchewan",
+    YUKON: "Yukon",
+}
+
+
+# scrape file formats
+def get_file_formats_from_site():
+    resp = make_request(SITE_URL)
+    tree = lxml.html.fromstring(resp.content)
+    list_item_path = (
+        "/html/body/main/div/div[3]/aside/div[2]/details[7]/ul/li[{}]/div/div[1]/label/div/text()"
+    )
+    file_formats = []
+    for i in range(100):
```

Comment:
assuming this is based on observations that there are nowhere near 100, but good to comment this assumption when using a hard-coded value